lib/gooddata/models/segment.rb

```diff
@@ -60,8 +63,11 @@ def all(opts = {})
         fail 'Domain has to be passed in options' unless domain
         client = domain.client
 
-        results = client.get(domain.segments_uri + '/segments')
-        GoodData::Helpers.get_path(results, %w(segments items)).map { |i| client.create(GoodData::Segment, i.merge('domain' => domain)) }
+        data_product = opts[:data_product]
+
+        uri = data_product ? GoodData::DataProduct::ONE_DATA_PRODUCT_PATH % { domain_name: domain.name, id: data_product.data_product_id } : domain.segments_uri
```

Comment:
Refers to 'domain' more than self (maybe move it to another class?) (FeatureEnvy)